build/DownloadCFamilyPlugin/DownloadAndExtract.cs

```diff
@@ -0,0 +1,266 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using SharpCompress.Compressors.Xz;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+
+namespace DownloadCFamilyPlugin
+{
+    /// <summary>
+    /// Downloads the CFamily jar and extracts the files that need to be
+    /// embedded in the SLVS vsix
+    /// </summary>
+    /// <remarks>
+    /// Assumptions:
+    /// * we are downloading the CFamily plugin jar e.g. https://binaries.sonarsource.com/CommercialDistribution/sonar-cfamily-plugin/sonar-cfamily-plugin-6.3.0.11371.jar
+    /// * the jar contains a .tarxz archive containing the Windows version of SubProcess.exe
+    /// * the jar contains various .json files, some with fixed names, some that we'll match using wildcards
+    /// 
+    /// We want the task to fail if it cannot locate all of the expected files (otherwise we'll build an invalid VSIX).
+    /// 
+    /// Downloading and extracting the files can be slow so we want to skip those steps if possible.
+    /// </remarks>
+    public class DownloadAndExtract: Task
+    {
+        // The txz archive containing the subprocess.exe
+        private const string WindowsTxzFilePattern = "clang-*-win.txz";
+        private const string CLangExeFilePattern = "SubProcess.exe";
+
+        private readonly string[] KnownJsonFiles = new string[]
+        {
+                "Sonar_way_profile.json",
+                "RulesList.json"
+        };
+
+        private readonly string[] AdditionalFilesToReturnPattern = new string[]
+        {
+            "S*_params.json"
```

Comment:
Not all rules starts with 'S': https://github.com/SonarSource/sonar-cpp/tree/master/cpp-checks/src/main/resources/org/sonar/l10n/cpp/rules/params